### PR TITLE
fix compilation errors

### PR DESCRIPTION
**Pull request by PaulDotSH**
Fixes the compilation errors on new rust version

### CODE REVIEW

**Comment by PaulDotSH** [comment]
@vorot93 @Boog900 @h4sh3d

File: src/util/amount.rs [hunk 1]
```diff
@@ -1085,7 +1085,7 @@ pub mod serde {
                     where
                         S: de::SeqAccess<'de>,
                     {
-                        let mut amt_vec = Vec::with_capacity(size_hint::cautious(seq.size_hint()));
+                        let mut amt_vec = Vec::with_capacity(size_hint::cautious::<usize>(seq.size_hint()));
```
Comment (Boog900) [inline src/util/amount.rs:1088]: this should be `::<Amount>`

this applies to the other usages of size_hint

**Review by Boog900 — CHANGES_REQUESTED** [review]
There are already 2 open PRs that fix this issue.

As stated in the other 2 PRs, I feel we should move away from serdes `__private` completely.

File: src/util/amount.rs (hunk 1 above)
Comment (PaulDotSH) [reply src/util/amount.rs:1088]: > this should be `::<Amount>`
> 
> this applies to the other usages of size_hint

Sorry, I didn't see them, I'll close this then, also can you please check on the monero-rpc-rs issues I made?

**Pull request closed without merging**